File: ssc/apiwrappers/java/SSC/Entry.java
package SSC;

public class Entry 
{
    private long m_entry;
    private int m_idx;

    public Entry()
    {
        m_idx = 0;
    }

    public void reset()
    {
        m_idx = 0;
    }

    public boolean get()
    {
        m_entry = SSC.SSCAPIJNI.ssc_module_entry(m_idx);
        if (m_entry == 0)
        {
            reset();
            return false;
        }

        m_idx++;
        return true;
    }

    public String name()
    {
        if (m_entry != 0)
            return SSC.SSCAPIJNI.ssc_entry_name(m_entry);
        else 
            return null;
    }

    public String description()
    {
        if (m_entry != 0)
            return SSC.SSCAPIJNI.ssc_entry_description(m_entry);
        else
            return null;
    }

    public int version()
    {
        if (m_entry != 0)
            return SSC.SSCAPIJNI.ssc_entry_version(m_entry);
        else
            return -1;
    }
    
}

File: ssc/apiwrappers/matlab/+SSC/@Entry/Entry.m
classdef Entry < handle
    %Entry object from SSC
    %   Object used to gather ssc module name, description and version
    %   information.
    
    properties (SetAccess = private)
        m_idx
        m_entry
    end
    properties
        Name
        Description
        Version
    end
    
    methods
        function obj = Entry()
            obj.m_idx = 0;
        end
        function Reset(obj)
            obj.m_idx = 0;
        end
        function result = Get(obj)
            obj.m_entry = SSC.ssccall('module_entry', obj.m_idx);
            if (obj.m_entry == 0),
                obj.Reset;
                result = false;
            else
                obj.Name = SSC.ssccall('entry_name', obj.m_entry);
                obj.Description = SSC.ssccall('entry_description', obj.m_entry);
                obj.Version = SSC.ssccall('entry_version', obj.m_entry);
                obj.m_idx = obj.m_idx + 1;
                result = true;
            end
        end
    end
    
end

